fastq/singleCell.go

```diff
@@ -63,8 +63,8 @@ func UmiCollision(sc SingleCellPair, umiMap map[string]int) bool {
 	count, found := umiMap[currUmi]
```

Comment:
Plan:
- run go profiler to see what function takes up lots of memory/time
- maybe a better way to extract barcode (cell barcode, UMI)
- bigger change: because map might be big, might be able to sort fastq (use efficient code in gonomics), and have less look-up (different way to count cell barcodes and UMI)